File: DSA/linked_list/sumList.js
class Node {
	constructor(val) {
		this.val = val
		this.next = null
	}
}

const a = new Node(2)
const b = new Node(8)
const c = new Node(3)
const d = new Node(-1)
const e = new Node(7)

a.next = b
b.next = c
c.next = d
d.next = e

const getSum = (node) => {
	let sum = 0
	while (node != null) {
		sum += node.val
		node = node.next
	}
	return sum
}

const getSumRec = (node) => {
	if (node == null) return 0
	return node.val + getSumRec(node.next)
}

console.log('Getting sum')
console.log(getSum(a))
console.log('Getting sum recursively')
console.log(getSumRec(a))

File: DSA/linked_list/sum_list.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

a = Node(2)
b = Node(8)
c = Node(3)
d = Node(-1)
e = Node(7)

a.next = b
b.next = c
c.next = d
d.next = e

def get_sum(node):
    s = 0
    while node:
        s += node.val
        node = node.next
    return s

def get_sum_rec(node):
    if node == None:
        return 0
    return node.val + get_sum_rec(node.next)

if __name__ == '__main__':
    print('Getting sum')
    print(get_sum(a))
    print('Getting sum recursively')
    print(get_sum_rec(a))
